Port Abductors

Zert0X opened the pull request:
Портированы похитители(Abductors) с ТГстейшена.

Умеют разговаривать между своей командой при помощи телепатии.
Знают обычный язык.

Имеется три подвида:
1. Scientist - учёный, типичный учёный...;
2. Agent - самая робастная тварь которую вы видели;
3. Onemanteam (Solo) - два по цене одного;

Из экипировки имеется:
1. Science tool - штука позволяющая маркировать цели для телепортации, а также сканировать внешность людей(может быть отсканированна повторно и удалённо через камеры);
2. Alien pistol - облучает радиацие, стреляет долго, заряжается ещё дольше;
3. Silencer - Отключает радио на N-е время.
4. Универсальная палка 9000(Alien Baton) - имеет 4 режима, probe, stun, sleep, restrain:
   4.1 Probe - просто проверяем можем ли мы вообще использовать эту тушу и не использована ли она уже? 
   4.2 Stun - даёт стан на 10 секунд;
   4.3 Sleep - укладывает в сон связанные тушки;
   4.4 Restrain - связывает тушку;
5. Alien vest - бронежилет имеющий два режима Stealth и Combat.
   5.1 В боевом режиме значительно повышается броня
   5.2 В стелс режиме доступна маскировка
6. Bail out implant - Только учёный и соло, позволяет вернуться на корабль
7. Также имеется пояс с набором инструментов.
8. Различная машинерия и хирургические инструменты(не забывайте что мы на бее и спользуйте перчатки и маску)

Имеется 13 желез далее "гланд", которые дают дополнительные возможности персонажу:
1. Access - фул доступ;
2. Blood - периодическое высасывание крови из окружающих;
3. Chem - периодически вводит в кровь различных химикаты.(аккуратно может убить);
4. Egg - поздравляю вы курица;
5. EMP - стань самым страшным сном синтов;
6. Heal - периодически восстанавливает повреждённые органы и конечности.(аккуратно может убить);
7. Metroid - станьте мамой метроидом!
8. Mindshock - вы слишком громко думаете из-за этого у остальных болит голова.
9. Плазма - ого халявный форон!?.(аккуратно может убить);
10. Quantum - познайте прелести квантовой неопределённости, может быть вы находитесь не там где думаете?
11. Transform - а вы точно тот кем вы себя считаете?
12. Ventcrawl - аналог костюма БДСМщика.
13. Viral - СOVID-13 (аккуратно может убить);

При имплантации гланды человеку выдаётся случайная задача для роли abductee.
Список задач: 
1. Украдите все(х) питомцев/лампочки/обезьян/фрукты/ботинки/куски мыла/оружие/компьютеры/органы
2. Станция отвратительна перекрасьте её в <цвет>/кровь
3. Ваш мозг сломан вы можете общаться только: пантомимой, рифмой, хайку и т.д.
4. Захватите <профессия>/кого-то
5. Вызовите призрака <имя_госта> на спиритический сеанс
6. Игнорируйте всё в количестве <2-10> этого не существует!
7. Список остальных в файле config\example\objectives\abductee_objectives.txt;

Есть шапочка из фольги! Частично защищает от воздействия антага.

<details>
<summary>Чейнджлог</summary>

```yml
🆑
rscadd: Портированы похитители!
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Zert0X committed:
abductor icons and some sounds

Zert0X committed:
abductee & abductee objectives

Zert0X committed:
abductor datums

Zert0X committed:
remove absence of hit reaction

Zert0X committed:
Abductors gear

Zert0X committed:
Create abductors_outfits.dm

Zert0X committed:
Adding glands

Zert0X committed:
Update access.dm

Zert0X committed:
Abductor Camera

Zert0X committed:
Abductor chem dispenser

Zert0X committed:
Abductor Console

Zert0X committed:
Gland Dispenser

Zert0X committed:
Experiment machine

Zert0X committed:
Abductor teleporter pad

Zert0X committed:
abductors machine

Zert0X committed:
TGUI interface

Zert0X committed:
Adding gland to ventcrawl list

Zert0X committed:
GLAND HUD

Zert0X committed:
Abductor species

Zert0X committed:
Update metroid_AI.dm

Zert0X committed:
Fix restore_organ

YEP THAT'S WAS BROKEN

Zert0X committed:
Escaping from experiment machine

Zert0X committed:
Add stump rejuvenate

Zert0X committed:
TINFOIL HAT

Big brother is watching you!

Zert0X committed:
Defines tweaks

Zert0X committed:
innate action type

Zert0X committed:
adding icon_snapshot

Zert0X committed:
Abductor ship area

Zert0X committed:
Abductor implant

Zert0X committed:
abductor walls and floors

Zert0X committed:
Abductors to maps

Zert0X committed:
Add abductors communication

Zert0X committed:
Abductor Headset

Zert0X committed:
Update baystation12.dme

NefrNerf commented:
Го без теста в прод

Zert0X committed:
Create abductee_objectives.txt

Zert0X commented:
> Го без теста в прод

Я только за, оно по идее должно почти полностью работать. Правда если педали будут спавнить абдукторов вручную :kek:

TobyThorne commented on code/game/antagonist/outsider/abductor/abductee/abductee.dm:
```diff
@@ -0,0 +1,26 @@
+
+/datum/antagonist/abductee/greet(datum/mind/player)
+	to_chat(player.current, SPAN_WARNING("<b>Your mind snaps!</b>"))
+	to_chat(player.current, "<big>[SPAN_WARNING("<b>You can't remember how you got here...</b>")]</big>")
```
почему не 
SPAN("warning", "\<big>\<b>You can't remember how you got here...\</b>\</big>")

TobyThorne commented on code/game/antagonist/outsider/abductor/equipment/abductors_gear.dm:
```diff
@@ -0,0 +1,830 @@
+/obj/item/paper/guides/antag/abductor
+	name = "Dissection Guide"
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "alienpaper_words"
```
дримчекер говорит, спрайта нема

TobyThorne commented on code/game/antagonist/outsider/abductor/equipment/abductors_gear.dm:
```diff
@@ -0,0 +1,830 @@
+/obj/item/gun/energy/abductor/alien
+	name = "alien pistol"
+	desc = "A complicated gun that fires bursts of high-intensity radiation."
+	icon_state = "alienpistol"
```
как и здесь

TobyThorne commented on code/game/antagonist/outsider/abductor/equipment/abductors_gear.dm:
```diff
@@ -0,0 +1,830 @@
+			handcuffs.place_handcuffs(C, user)
+			return
+		else
+			to_chat(user, "<span class='warning'>\The [C] is already handcuffed!</span>")
```
```suggestion
			to_chat(user, SPAN("warning", "\The [C] is already handcuffed!"))
```

TobyThorne commented on code/game/antagonist/outsider/abductor/equipment/abductors_gear.dm:
```diff
@@ -0,0 +1,830 @@
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		species = SPAN_NOTICE("[H.dna.species]")
+		if(H.mind && !isnull(H.mind.changeling))
```
```suggestion
		if(H.mind?.changeling)
```
не?

TobyThorne commented on code/modules/mob/living/carbon/human/human_organs.dm:
```diff
@@ -40,15 +40,14 @@
+		var/organ_path = species.has_organ[organ_type]
 		var/obj/item/organ/internal/O = new organ_path(src)
-		organ_data["descriptor"] = O.name
+		src.internal_organs_by_name[organ_type] = O
```
зачем src.

TobyThorne commented on code/modules/mob/living/living.dm:
```diff
@@ -667,11 +667,17 @@
 		return TRUE
 
+	//Trying to escape from abductors?
+	if(src.loc && (istype(src.loc, /obj/machinery/abductor/experiment)))
```
зачем src.

TobyThorne commented on code/modules/mob/living/living.dm:
```diff
@@ -667,11 +667,17 @@
 
 	//Breaking out of a locker?
-	if( src.loc && (istype(src.loc, /obj/structure/closet)) )
+	if(src.loc && (istype(src.loc, /obj/structure/closet)) )
```
зачем src.

TobyThorne commented on code/modules/organs/external/stump.dm:
```diff
@@ -20,3 +20,16 @@
+/obj/item/organ/external/stump/rejuvenate(ignore_prosthetic_prefs)
+	. = ..()
+	var/mob/living/carbon/human/H = owner
+	src.removed(drop_organ = FALSE)
```
зачем src.

TobyThorne reviewed (commented):
почитал, вроде норм, но чёт рука не поднимается на галочку
пусть кто-нибудь на локалке погоняет и убедится, что ничто не высирает тысячу рантаймов в секунду и не укладывает сервер

NefrNerf commented:
Да уже тестим, много серьезных багов.

Zert0X commented:
> Да уже тестим, много серьезных багов.

Ой да ладно тебе, не такие уж и серьёзные :kek:

Zert0X committed:
Change check_hit_reaction() to handle_shield()

Zert0X committed:
Some bugs fixed

Zert0X committed:
Balance

Zert0X committed:
Adding more aliens surgery tools

Zert0X committed:
Fixed gun icon

Zert0X committed:
Added tinfoil hat icon

Zert0X committed:
Update mob.dmi

Zert0X committed:
update maps

Zert0X committed:
Fixed ventcrawl

Zert0X committed:
Update human_attackhand.dm

Zert0X committed:
tweak tinfoil hat armor

Zert0X committed:
deleted hit_reaction

Zert0X committed:
Fixed some machinery bugs

- unreleaseable camera
- teleportation without cooldown
- giving Abductee role

Zert0X committed:
Fixed Gland organ tag

Zert0X committed:
Added spawn_req & spawn_target for antag datum

Zert0X committed:
Glands fixing and tweaking

- Rewrited blood actions
- Fixed egg reagent picking
- Fixed EMP gland desc
- Tweaked mindshock gland timings
- Tweaked plasma amount and CoolDown
- Tweaked quantum linking to abductors
- Fixed transform gland
- Fixed viral gland desc

Zert0X committed:
some github suggestions

Zert0X committed:
Fixed alienpistol icon_state

yep, that's doesn't mater but linter

Zert0X committed:
Added abductors spawn by storyteller

I AM HATE THIS SHIT PLS SMB TRY TO MAKE IT BETTER

Zert0X committed:
Alient pistol recharge time tweak

Zert0X committed:
Dissection Guide info update

igorsaux reviewed (changes requested):
when 
![image](https://user-images.githubusercontent.com/55196698/159944917-8b907914-5aab-4bfc-8db6-9ff6dcc245ff.png)

Zert0X commented:
> when ![image](https://user-images.githubusercontent.com/55196698/159944917-8b907914-5aab-4bfc-8db6-9ff6dcc245ff.png)

Листы оставим потому что так можно, последнее пофикшено.
Сейчас ещё доделываю смолфиксы всякого разного и можно в прод.

igorsaux commented:
Если бы было можно - тогда и проверки не было

Zert0X commented:
> Если бы было можно - тогда и проверки не было

вангую проверка на наличие пробела

Zert0X committed:
Glands fix

Zert0X committed:
Camera adv fix

Zert0X committed:
Linter fix

Zert0X committed:
Some tweaks

igorsaux reviewed (dismissed):
убрал каку

Zert0X committed:
linter
